src/main/java/io/confluent/connect/hdfs/wal/FSWAL.java

```diff
@@ -142,6 +143,10 @@ public void apply() throws ConnectException {
       log.error("Error applying WAL file '{}' because it is corrupted: {}", logFile, e);
       log.warn("Truncating and skipping corrupt WAL file '{}'.", logFile);
       close();
+    } catch (CannotObtainBlockLengthException e) {
+      log.error("Error applying WAL file '{}'as not able to obtain block length: {}", logFile, e);
+      log.warn("Truncating and skipping the WAL file '{}'.", logFile);
+      close();
     } catch (IOException e) {
```

Comment:
`CannotObtainBlockLengthException` is a subclass of `IOException` and is already addressed

Reply:
IOException will throw exception, while here we will continue

Reply:
I see, then it should be addressed in `commitWalEnteriesToStorage` where the corrupt wal file exception is thrown

Reply:
I agree with @yanglei99 that it's good to have a similar handling of this error next to how we handle corruption errors.